### PR TITLE
Carson/api fixes

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Description

Simplify authentication flow for remote gRPC APIs.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Should not affect existing tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

> This PR template comes from https://github.com/embeddedartistry/templates

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -47,6 +47,15 @@ export class Client {
     this.rpcOptions.transport && grpc.setDefaultTransport(this.rpcOptions.transport)
   }
 
+  /**
+   * Create a new gRPC client instance from a supplied user auth object.
+   * Assumes all default gRPC setttings. For custimization options, use a context object directly.
+   * @param auth The user auth object.
+   */
+  static withUserAuth(auth: UserAuth) {
```
Comment (andrewxhill) [inline packages/client/src/index.ts:55]: doh, this cannot be combined with a custom endpoint I don't think

Comment (carsonfarmer) [reply packages/client/src/index.ts:55]: Not sure what that means? But is it because it’s a static method?

Comment (andrewxhill) [reply packages/client/src/index.ts:55]: where would you include the host string?

Comment (carsonfarmer) [reply packages/client/src/index.ts:55]: Ah yeh see docs strings. I didn’t realize this was needed 😩. Easy fix, can add those as optional args as in other methods. I won’t be able to get to that right away though.

Comment (andrewxhill) [reply packages/client/src/index.ts:55]: yeah, no biggie right now

**Pull request merged**